x-pack/plugins/observability_solution/observability_ai_assistant/public/hooks/use_user_preferred_language.ts

```diff
@@ -0,0 +1,218 @@
+  mg: 'Malagasy',
+  mk: '\u043c\u0430\u043a\u0435\u0434\u043e\u043d\u0441\u043a\u0438',
+  ml: '\u0d2e\u0d32\u0d2f\u0d3e\u0d33\u0d02',
+  mn: '\u043c\u043e\u043d\u0433\u043e\u043b',
+  mr: '\u092e\u0930\u093e\u0920\u0940',
+  ms: 'Bahasa Melayu',
+  mt: 'Malti',
+  my: '\u1017\u1019\u102c',
+  nb: 'norsk bokm\u00e5l',
+  nd: 'isiNdebele',
+  ne: '\u0928\u0947\u092a\u093e\u0932\u0940',
+  nl: 'Nederlands',
+  nl_BE: 'Vlaams',
+  nn: 'nynorsk',
+  no: 'norsk',
+  om: 'Oromoo',
+  or: '\u0b13\u0b21\u0b3c\u0b3f\u0b06',
+  os: '\u0438\u0440\u043e\u043d',
+  pa: '\u0a2a\u0a70\u0a1c\u0a3e\u0a2c\u0a40',
+  pl: 'polski',
+  ps: '\u067e\u069a\u062a\u0648',
+  pt: 'portugu\u00eas',
+  pt_BR: 'portugu\u00eas do Brasil',
+  pt_PT: 'portugu\u00eas europeu',
+  qu: 'Runasimi',
+  rm: 'rumantsch',
+  rn: 'Ikirundi',
+  ro: 'rom\u00e2n\u0103',
+  ro_MD: 'moldoveneasc\u0103',
+  ru: '\u0440\u0443\u0441\u0441\u043a\u0438\u0439',
+  rw: 'Kinyarwanda',
+  se: 'davvis\u00e1megiella',
+  sg: 'S\u00e4ng\u00f6',
+  sh: 'Srpskohrvatski',
+  si: '\u0dc3\u0dd2\u0d82\u0dc4\u0dbd',
+  sk: 'sloven\u010dina',
+  sl: 'sloven\u0161\u010dina',
+  sn: 'chiShona',
+  so: 'Soomaali',
+  sq: 'shqip',
+  sr: '\u0441\u0440\u043f\u0441\u043a\u0438',
+  sv: 'svenska',
+  sw: 'Kiswahili',
+  ta: '\u0ba4\u0bae\u0bbf\u0bb4\u0bcd',
+  te: '\u0c24\u0c46\u0c32\u0c41\u0c17\u0c41',
+  th: '\u0e44\u0e17\u0e22',
+  ti: '\u1275\u130d\u122d\u129b',
+  tl: 'Tagalog',
+  to: 'lea fakatonga',
+  tr: 'T\u00fcrk\u00e7e',
+  ug: '\u0626\u06c7\u064a\u063a\u06c7\u0631\u0686\u06d5',
+  uk: '\u0443\u043a\u0440\u0430\u0457\u043d\u0441\u044c\u043a\u0430',
+  ur: '\u0627\u0631\u062f\u0648',
+  uz: 'o\u02bbzbekcha',
+  vi: 'Ti\u1ebfng Vi\u1ec7t',
+  yi: '\u05d9\u05d9\u05b4\u05d3\u05d9\u05e9',
+  yo: '\u00c8d\u00e8 Yor\u00f9b\u00e1',
+  zh: '\u4e2d\u6587',
+  zh_Hans: '\u7b80\u4f53\u4e2d\u6587',
+  zh_Hant: '\u7e41\u9ad4\u4e2d\u6587',
+  zu: 'isiZulu',
+};
+
+const LANGUAGE_OPTIONS = SPECIAL_OPTIONS.concat(
+  Object.values(languages).map((language) => ({
+    label: language,
+  }))
+);
+
+export type UseUserPreferredLanguageResult = ReturnType<typeof useUserPreferredLanguage>;
+
+export const SELECTED_LANGUAGE_LOCAL_STORAGE_KEY =
+  'xpack.observabilityAiAssistant.responseLanguage';
+
+export function useUserPreferredLanguage() {
+  const [selectedLanguage, selectLanguage] = useLocalStorage(
+    SELECTED_LANGUAGE_LOCAL_STORAGE_KEY,
+    USE_KIBANA_LOCALE_SETTING
+  );
+
+  const getPreferredLanguage = useCallback(() => {
+    if (selectedLanguage === USE_KIBANA_LOCALE_SETTING) {
+      return getLanguageFromLocale();
+    } else if (selectedLanguage === USE_BROWSER_LANGUAGE_SETTING) {
+      return getLanguageFromBrowserSetting();
+    } else {
+      return selectedLanguage || 'English';
+    }
+  }, [selectedLanguage]);
+
+  return {
+    selectedLanguage,
+    selectLanguage,
+    LANGUAGE_OPTIONS,
+    getPreferredLanguage,
+  };
+}
+
+function getLanguageFromLocale() {
+  switch (i18n.getLocale()) {
```

Comment:
Where does this setting actually come from and how is it changed? Is it from Kibana settings? If so, perhaps call it `getLanuageFromKibanaSettings`